src/persist/src/location.rs

```diff
@@ -373,6 +412,35 @@ impl From<(&str, &str)> for LockInfo {
     }
 }
 
+/// An abstraction over read-write access to a `bytes key`->`bytes value` store.
```

Comment:
small nit: we might want to take a bit of time to say what consistency model we expect the implementation to follow. Like, concretely, we've talked about how we don't need the blob implementation to have linearizability because everything is write once and readers can retry. I _think_ this means that the blob store can be eventually consistent and we don't have any other requirements on it but worth double checking.

Reply:
done, but made it linearizable to start with a TODO to see if we can relax it (see the TODO for why). the blob_multi_test I just added certainly assumes it's linearizable

is that reasoning compelling?